WebHostLib/static/assets/player-options.js

```diff
@@ -1,523 +1,499 @@
+          if (romOpts) { select.setAttribute('data-romOpt', '1'); }
+          options[option].options.forEach((opt) => {
+            const optionElement = document.createElement('option');
+            optionElement.setAttribute('value', opt.value);
+            optionElement.innerText = opt.name;
+
+            if ((isNaN(this.current[option]) &&
+              (parseInt(opt.value, 10) === parseInt(this.current[option]))) ||
+              (opt.value === this.current[option]))
+            {
+              optionElement.selected = true;
+            }
+            select.appendChild(optionElement);
+          });
+          select.addEventListener('change', (event) => this.#updateGameOption(event.target));
+          element.appendChild(select);
+
+          // Randomize button
+          randomButton.innerText = '🎲';
+          randomButton.classList.add('randomize-button');
+          randomButton.setAttribute('data-key', option);
+          randomButton.setAttribute('data-tooltip', 'Toggle randomization for this option!');
+          randomButton.addEventListener('click', (event) => toggleRandomize(event, select));
+          if (this.current[option] === 'random') {
+            randomButton.classList.add('active');
+            select.disabled = true;
+          }
 
-      default:
-        console.error(`Ignoring unknown option type: ${options[option].type} with name ${option}`);
-        return;
-    }
+          element.appendChild(randomButton);
+          break;
+
+        case 'range':
+          element = document.createElement('div');
+          element.classList.add('range-container');
+
+          let range = document.createElement('input');
+          range.setAttribute('id', option);
+          range.setAttribute('type', 'range');
+          range.setAttribute('data-key', option);
+          range.setAttribute('min', options[option].min);
+          range.setAttribute('max', options[option].max);
+          range.value = this.current[option];
+          range.addEventListener('change', (event) => {
+            document.getElementById(`${option}-value`).innerText = event.target.value;
+            this.#updateGameOption(event.target);
+          });
+          element.appendChild(range);
+
+          let rangeVal = document.createElement('span');
+          rangeVal.classList.add('range-value');
+          rangeVal.setAttribute('id', `${option}-value`);
+          rangeVal.innerText = this.current[option] !== 'random' ?
+            this.current[option] : options[option].defaultValue;
+          element.appendChild(rangeVal);
+
+          // Randomize button
+          randomButton.innerText = '🎲';
+          randomButton.classList.add('randomize-button');
+          randomButton.setAttribute('data-key', option);
+          randomButton.setAttribute('data-tooltip', 'Toggle randomization for this option!');
+          randomButton.addEventListener('click', (event) => toggleRandomize(event, range));
```

Comment:
This and all the later calls to `toggleRandomize` should be `this.#toggleRandomize`